# Default fake yolo launch for leviathan
# generates bounding boxes similar to yolo for testing without GPU grunt, or before network is trained
yolo_faker:
 bounding_boxes_topic: 'cusub_perception/darknet_ros/bounding_boxes'
 sub_pose_topic: 'description/pose_gt'
 show_points: True

# Cameras to run fake yolo on
 cameras:
   occam0:
     frame: 'leviathan_gt/occam0_frame'
     image_topic: 'cusub_common/occam/image0'
     debug_image_enabled: True
     debug_image_topic: 'cusub_common/occam/image0_debug'
     camera_info_topic: 'cusub_common/occam/image0/camera_info'
   downcam:
     frame: 'leviathan_gt/downcam_frame'
     image_topic: 'cusub_common/downcam'
     debug_image_enabled: True
     debug_image_topic: 'cusub_common/downcam_debug'
     camera_info_topic: 'cusub_common/downcam/camera_info'

# Objects to run fake yolo on
# You can have multiple classes per object with different point offsets, like the 2 start gate poles
 objects:
  # startgate:
  #  link: 'StartGate_1/base_link'
  #  odom_topic: '/StartGate_1/pose_gt'
  #  classes:
  #   - name: 'start_gate_pole'
  #     debug_color: [0, 255, 0]
  #     points: [[ 1.450, -0.075,  0.000],[ 1.650, -0.075,  0.000],[ 1.450,  0.075,  0.000],[ 1.650,  0.075,  0.000],[ 1.450, -0.075, -1.500],[ 1.650, -0.075, -1.500],[ 1.450,  0.075, -1.500],[ 1.650,  0.075, -1.500]]
  #   - name: 'start_gate_pole'
  #     debug_color: [0, 255, 0]
  #     points: [[-1.450, -0.075,  0.000],[-1.650, -0.075,  0.000],[-1.450,  0.075,  0.000],[-1.650,  0.075,  0.000],[-1.450, -0.075, -1.500],[-1.650, -0.075, -1.500],[-1.450,  0.075, -1.500],[-1.650,  0.075, -1.500]]
  #   - name: 'start_gate_pole'
  #     debug_color: [0,255,0]
  #     points: [[ 0.430, -0.075, 0.000],[ 0.250, -0.075, 0.000],[0.430, 0.075,0.000],[0.250,0.075,0.000],[ 0.430, -0.075, -0.750],[ 0.250, -0.075, -0.750],[0.430, 0.075, -0.750],[0.250,0.075, -0.750]]
  # path0:
  #  link: 'PathMarker_1/base_link'
  #  odom_topic: '/PathMarker_1/pose_gt'
  #  classes:
  #   - name: 'path'
  #     debug_color: [0, 255, 0]
  #     points: [ [0.01, 0.01, 0.01], [-0.01, 0.01, 0.01], [-0.01, -0.01, 0.01], [-0.01, 0.01, -0.01], [0.01, -0.01, 0.01], [0.01, -0.01, -0.01], [0.01, 0.01, -0.01], [-0.01, -0.01, -0.01] ]
  # jiangshi:
  #  link: 'buoy_jiangshi_1/base_link'
  #  odom_topic: '/buoy_jiangshi_1/pose_gt'
  #  classes:
  #   - name: 'vampire_cute'
  #     debug_color: [0, 255, 0]
  #     points: [ [-0.4048, -0.0508, 0.6096], [ 0.4048, -0.0508,  0.6096], [ -.4048,  0.0508,  0.6096], [ 0.4048,  0.0508,  0.6096], [-0.4048, -0.0508, -.6096], [ 0.4048, -0.0508,  -.6096], [ -.4048,  0.0508,  -.6096], [ 0.4048,  0.0508,  -.6096] ]
  triangle:
    link: 'triangular_buoy_1/base_link'
    odom_topic: 'triangular_buoy_1/pose_gt'
    classes:
      - name: 'vampire_flying'
        debug_color: [200,200,0]
        points: [[.5,0,-1],[0,.5,0],[.5,0,0],[.5,.5,-1]]
      - name: 'vampire_fathead'
        debug_color: [0,200,200]
        points: [[-.5,0,0],[0,-.5,0],[-.5,-.5,0],[0,0,0]]
      - name: 'vampire_no_torso'  
        debug_color: [200,0,200]
        points: [[.4,-.4,0],[0,-.4,0],[.4,0,0],[.4,.4,0]]  
  # dropper:
  #  link: 'dropper_garlic_1/base_link'
  #  odom_topic: '/dropper_garlic_1/pose_gt'
  #  classes:
  #   - name: 'dropper_cover'
  #     debug_color: [255, 0, 0]
  #     points: [[-0.05,.53,0.1],[.4,.53,0.1],[-0.05,1.1,0.1],[.4,1.1,0.1]]
  #   - name: 'wolf'
  #     debug_color: [0,0,255]
  #     points: [[.72,.55,-0.1],[1,.55,-.1],[.72,1.09,-.1],[1,1.09,-0.1]]
  # torpedo:
  #  link: 'vampire_stake_1/base_link'
  #  odom_topic: 'vampire_stake_1/pose_gt'
  #  classes:
  #   - name: 'dracula'
  #     debug_color: [50,150,200]
  #     points: []
  #   - name: 'open_oval'
  #     debug_color: [150,200,50]
  #     points: []
  #   - name: 'heart'
  #     debug_color: [200,150,50]
  #     points: []
  # coffin:
  #  link: 'coffin_1/base_link'
  #  odom_topic: 'coffin_1/pose_gt'
  #  classes:
  #   - name: 'coffin'
  #     debug_color: [20,40,70]
  #     points: []
